File: SDKs/parliament.uk/oralquestions/v1/python/src/sdk/models/shared/paginginfo.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class PagingInfo:
    skip: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Skip') }})
    status_counts: Optional[List[StatusCount]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('StatusCounts') }})
    take: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Take') }})
    total: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Total') }})
    

File: SDKs/parliament.uk/oralquestions/v1/typescript/src/sdk/models/shared/paginginfo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { StatusCount } from "./statuscount";



export class PagingInfo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Skip" })
  skip?: number;

  @SpeakeasyMetadata({ data: "json, name=StatusCounts", elemType: StatusCount })
  statusCounts?: StatusCount[];

  @SpeakeasyMetadata({ data: "json, name=Take" })
  take?: number;

  @SpeakeasyMetadata({ data: "json, name=Total" })
  total?: number;
}
